test/integration/ads_integration_test.cc

```diff
@@ -487,7 +487,7 @@ TEST_P(AdsIntegrationTest, CdsEdsReplacementWarming) {
       {buildTlsCluster("cluster_0")}, {}, "2");
   // Inconsistent SotW and delta behaviors for warming, see
   // https://github.com/envoyproxy/envoy/issues/11477#issuecomment-657855029.
-  if (sotw_or_delta_ != Grpc::SotwOrDelta::Delta) {
+  if (sotw_or_delta_ == Grpc::SotwOrDelta::Sotw) {
```

Comment:
Please add a TODO next to the places where only non-unified SotW has a unique behavior, so it would be easier to understand that these code sections should be removed once the legacy code is removed.